backend/infrahub/task_manager/event.py

```diff
@@ -192,8 +193,18 @@ async def query_events(
     ) -> PrefectEventResponse:
         body = {"limit": limit, "filter": filters.model_dump(mode="json", exclude_none=True), "offset": offset}
 
-        response = await client._client.post("/infrahub/events/filter", json=body)
-        response.raise_for_status()
+        # Retry due to https://github.com/PrefectHQ/prefect/issues/16299
+        attempt = 1
+        retry = True
+        while retry:
```

Comment:
I think it would be easier to read and understand with a for loop and a counter with the maximum number of retry
if the response is good we can just return from within the loop

Also if the server is still unreachable after N attempts, it would be good to generate our own exception a 503 or 504 would be better than a 500

Reply:
I've made some updates to this and introduced a custom error.